internal/storage/s3backend/client.go

```
@@ -0,0 +1,312 @@
+func (c *client) Download(ctx context.Context, key string) ([]byte, error) {
+	result, err := c.s3.GetObject(ctx, &s3.GetObjectInput{
+		Bucket: aws.String(c.cfg.Bucket),
+		Key:    aws.String(key),
+	})
+	if err != nil {
+		return nil, ierr.WithError(err).
+			WithHint("failed to download object from S3").
+			WithMessagef("bucket:%s, key:%s", c.cfg.Bucket, key).
+			Mark(ierr.ErrHTTPClient)
+	}
+	defer result.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `result.Body.Close` is not checked

(errcheck)